src/main/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactVersionScore.java

```diff
@@ -0,0 +1,80 @@
+package com.sap.oss.phosphor.fosstars.model.score.oss;
+
+import static com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures.ARTIFACT_VERSION;
+import static com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures.RELEASED_ARTIFACT_VERSIONS;
+
+import com.sap.oss.phosphor.fosstars.model.Score;
+import com.sap.oss.phosphor.fosstars.model.Value;
+import com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures;
+import com.sap.oss.phosphor.fosstars.model.score.FeatureBasedScore;
+import com.sap.oss.phosphor.fosstars.model.value.ArtifactVersion;
+import com.sap.oss.phosphor.fosstars.model.value.ArtifactVersions;
+import com.sap.oss.phosphor.fosstars.model.value.ScoreValue;
+import java.time.LocalDate;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * The scores check how old the given artifact version is
+ * and if it is the latest available artifact version.
+ * <ul>
+ *   <li>{@link OssFeatures#RELEASED_ARTIFACT_VERSIONS}</li>
+ *   <li>{@link OssFeatures#ARTIFACT_VERSION}</li>
+ * </ul>
+ */
+public class ArtifactVersionScore extends FeatureBasedScore {
+
+  /**
+   * Initializes a new score.
+   */
+  public ArtifactVersionScore() {
+    super("How old the given version is and is it the latest available artifact version",
```

Comment:
@mibo

Reply:
IMHO we should keep the _and is it the latest available artifact version_ as this is checked together with the age of the version.

Reply:
The purpose of the name is not to describes how the scoring function does something. Instead, it should describe what it tries to score (not how). Looking at the `calculate()` method, I see that the newer version is, the higher the score. It makes me think the the scoring function assess how old the version is. Therefore I am suggesting to rename it. In the javadoc/description you can describe the logic in detail.

Reply:
Somehow I not really get it, sorry.
How old a version is does not relate to if it is the latest version.
So, two different artifacts for which A's version is a month old and is latest version and B's version is a month old but is not latest will get different scores.
If only the age of the version is relevant the scores should be the same.
What do I miss here?

Reply:
> How old a version is does not relate to if it is the latest version.

The latest version is always the youngest one. 

> So, two different artifacts for which A's version is a month old and is latest version and B's version is a month old but is not latest will get different scores.

One of them has to be older. Even if they both are one month old, only one of them (B) is younger anyway. For example, A is one hour older. Since the scoring function doesn't take into account hours, it checks if one of the versions is the latest one to assign the max score for it. If release date had hours/seconds/minutes/etc, it would not need to know which version is the latest one. It is all about age of versions.

Reply:
> How old a version is does not relate to if it is the latest version.

The latest version is always the youngest one. 

> So, two different artifacts for which A's version is a month old and is latest version and B's version is a month old but is not latest will get different scores.

One of them has to be older. Even if they both are one month old, only one of them (B) is younger anyway. For example, A is one hour older. Since the scoring function doesn't take into account hours, it checks if one of the versions is the latest one to assign the max score for it. If release date had hours/seconds/minutes/etc, it would not need to know which version is the latest one. It is all about age of versions.

Reply:
Sorry, that I still not get it.
Maybe I explained it wrong. Let's try a last time:
- Artifact A: 1.0.0 released 100 days ago, 1.1.0 released 10 days ago, 
- Artifact B: 1.0.0 released 100 days ago

ArtifactVersionScore for Artifact A in version 1.0.0 => e.g. score of 7
ArtifactVersionScore for Artifact B in version 1.0.0 => e.g. score of 9
Both artifact versions 1.0.0 have the same age (100days) but for artifact B it is also the latest release (no newer version available).

Reply:
Scores for artifacts A and B can't be compared. A has the latest version, and B has the latest version. These versions don't have to match, and they can't be compared since they may be released separately. We don't currently have artifact names (we've discussed it, and agreed to open an issue for this).

Reply:
Okay, I didn't explain it well enough. Please ignore my comment above. I'll try again.

Reply:
> Scores for artifacts A and B can't be compared.

This is wrong. Scores can be compared.

Here is what I mean. The score uses `RELEASED_ARTIFACT_VERSIONS`. When `calculate()` is called, this feature contains versions for a single artifact A. We don't consider any other artifact when we calculate a score here. One of the versions is the latest version. The latest version is always the younger one for this artifact. The latest versions gets the max score. Older versions get lower scores. Again, here we always compare ages of versions of the same artifact.

We can apply this scoring functions to another artifact B. Info about A will not be taken into account.

If we get scores for a versions of A, and for a version of B, we can not tell which one is better. That's why I said we can not compare the scores here. We can compare scores of the same artifact.

Reply:
For me it is a bit different.
As not only `RELEASED_ARTIFACT_VERSIONS` but also `ARTIFACT_VERSION` is considered.

As in my sample above, if `ARTIFACT_VERSION` is the latest version you will get a better score even it is older compared to the case that if `ARTIFACT_VERSION` is not the latest version but has the same age.

However, I not really get it but, as you are the expert for this topic and the library, I will change the name as suggested.

Reply:
Two versions of the same artifact can't have the same age. They may be 100 days old, but one can be, for example, 1 hour older.

I am not the expert here. I am just trying to understand what the score is doing. Then we can find a good short/long description. 

Looks like we don't understand each other. Let's please have a call tomorrow.

Reply:
@artem-smotrakov What do you think about _How up-to-date the given version is_ instead of _How old the given version is_ ?

Reply:
Sure, let's use it.